numpy/core/include/numpy/ndarraytypes.h

```diff
@@ -172,10 +172,12 @@ typedef enum {
 #define NPY_NSORTS (NPY_STABLESORT + 1)
 
 
+
 typedef enum {
-        NPY_INTROSELECT=0
+        NPY_INTROSELECT=0,
+        NPY_FLOYDSELECT=1
 } NPY_SELECTKIND;
-#define NPY_NSELECTS (NPY_INTROSELECT + 1)
+#define NPY_NSELECTS (NPY_FLOYDSELECT + 1)
```

Comment:
I think @hameerabbasi's point is that changing `NPY_NSELECTS` from `1` to `2` breaks the ABI.

Reply:
Thank you! Do you have any suggestions how to fix this? I've not worked with versioned libraries before and don't know how it's usually done.

Reply:
I may be wrong, but the C-API for this doesn't appear to be public. The structs are hidden. I think just the addition of a release note saying "we only support floydselect above this version, otherwise the C-API will crash" would be enough.